src/lib.rs

```diff
@@ -68,54 +79,127 @@
 //! and the various `structopt` attributes are simply
 //! used for additional parameters.
 //!
-//! First, define a struct, whatever its name.  This structure will
-//! correspond to a `clap::App`.  Every method of `clap::App` in the
-//! form of `fn function_name(self, &str)` can be used through attributes
-//! placed on the struct. In our example above, the `about` attribute
-//! will become an `.about("An example of StructOpt usage.")` call on the
-//! generated `clap::App`. There are a few attributes that will default
-//! if not specified:
-//!
-//!   - `name`: The binary name displayed in help messages. Defaults
-//!      to the crate name given by Cargo.
-//!   - `version`: Defaults to the crate version given by Cargo.
-//!   - `author`: Defaults to the crate author name given by Cargo.
-//!   - `about`: Defaults to the crate description given by Cargo.
-//!
-//! This approach also works for other literals or expressions which
-//! can be used as values of the attributes for corresponding
-//! `clap::App` methods (e.g. for `fn function(self, bool)` and so
-//! on).
-//!
-//! Then, each field of the struct not marked as a subcommand
-//! corresponds to a `clap::Arg`. As with the struct attributes, every
-//! method of `clap::Arg` with one argument can be used through
-//! specifying it as an attribute with a value which is either a
-//! literal or an expression. The `name` attribute can be used to
-//! customize the `Arg::with_name()` call (defaults to the field name
-//! in kebab-case).
-
-//! Some of the rare functions that take more than one argument can be
-//! specified as function calls directly, e.g. `required_if("out",
-//! "file")`. A full example:
-//!
+//! First, define a struct, whatever its name. This structure
+//! corresponds to a `clap::App`, its fields correspond to `clap::Arg`
+//! (unless they're [subcommands](#subcommands)),
+//! and you can adjust these apps and args by `#[structopt(...)]` attributes.
+//!
+//! `#[structopt(...)]` attributes fall into two categories:
+//! - `structopt`'s own [magical methods](#magical-methods), they are used by `structopt` itself.
+//!   They come mostly in `attr = ""` form, but some `attr(args...)` also exist.
+//! - [`raw` attributes](#raw-methods) (explicit `#[structopt(raw(...))]` attrs in
+//!   pre-0.3 `structopt`). They represent explicit `clap::Arg/App` method calls.
+//!
+//! Every `structopt attribute` looks like comma-separated sequence of methods:
+//! ```rust,ignore
+//! #[structopt(
+//!     short, // method with no arguments
```

Comment:
This one is magical as is generate the needed argument.

Reply:
Yep, fixed